recipes/enchant/all/patches/0003-fix-file-locking.patch

```diff
@@ -0,0 +1,100 @@
+--- src/pwl.c
++++ src/pwl.c
+@@ -54,14 +54,24 @@
+ 
+ #include "config.h"
+ 
++#define _DEFAULT_SOURCE 1
++
+ #include <stdio.h>
+ #include <stdlib.h>
+ #include <string.h>
+ #include <sys/types.h>
+ #include <sys/stat.h>
+-#include <sys/file.h>
+ #include <fcntl.h>
+ 
++#if defined _WIN32
++# include <io.h>
++# include <Windows.h>
++#elif defined HAVE_FLOCK
++# include <sys/file.h>
++#else
++# include <unistd.h>
++#endif
++
+ #include <glib.h>
+ #include <glib/gstdio.h>
+ #include "enchant-provider.h"
+@@ -182,8 +192,69 @@
+ 
+ static int edit_dist(const char* word1, const char* word2);
+ 
+-#define enchant_lock_file(f) flock (fileno (f), LOCK_EX)
+-#define enchant_unlock_file(f) flock (fileno (f), LOCK_UN)
++static void enchant_lock_file(FILE * f)
```

Comment:
it almost looks good, expect it is void and doesn't return status code or set errno (as an original [flock](https://linux.die.net/man/2/flock) does). it only prints a warning in case file couldn't be locked.
I don't know how is it used in the codebase, e.g. if someone is actually checking `errno` after `enchant_lock_file` calls.

Reply:
The function that existed before the macro did the same thing: https://github.com/AbiWord/enchant/commit/edf662a5ec8db85aa6e5a5d00531ab6b8773131f#diff-a54665f3f5d38281ae6f02c4e410a88d332d5b9f42466f05c35246ea23cd201dL196